File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    let reports = [...new Set(report)].map(a=>{return a.split(' ')});
    let dec_counts = new Map();
    let mail_counts = new Map();
    
    for (const report of reports){
        dec_counts.set(report[1],dec_counts.get(report[1])+1||1)
    }
    
    for(const report of reports){
        if(dec_counts.get(report[1])>=k){
            mail_counts.set(report[0],mail_counts.get(report[0])+1||1)
        }
    }
    
    return id_list.map(v => mail_counts.get(v)||0);
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = []
    reported_user = {}
    count = {}
    
    for r in report:
        reporter, reported = r.split( )
        if reported not in reported_user: reported_user[reported] = set()
        reported_user[reported].add(reporter)
        
    for reported, reporters in reported_user.items():
        if len(reporters) >= k:
            for uid in reporters:
                if uid not in count: count[uid] = 1
                else: count[uid] += 1
                
    for i in range(len(id_list)):
        if id_list[i] not in count: answer.append(0)
        else: answer.append(count[id_list[i]])
        
    return answer